lib/mjsonwp/mjsonwp.js

```diff
@@ -245,6 +245,11 @@ function buildHandler (app, method, path, spec, driver, isSessCmd) {
         driverRes = null;
       }
 
+      // if the status is not 0,  throw the appropriate error for status code.
+      if (util.hasValue(driverRes) && util.hasValue(driverRes.status) && parseInt(driverRes.status, 10) !== 0) {
+        throw errorFromCode(driverRes.status, driverRes.value);
```

Comment:
I think we still need to settle on the overall design; the current design feels wrong to me. What do you think @imurchie